### DISCUSSION TITLE
httpx instead of requests and aiohttp

### DISCUSSION THREAD

**Discussion by T-256** [eternnoir/pyTelegramBotAPI#2016] [Ideas]
use less packages (and perf improvements?)

**Comment by coder2020official** [comment]
Hello, sorry for late response. The library was first built on top of requests, so therefore making hazardous changes would take us lots of time, and as a result, we had to choose: either rewriting everything from scratch using a library that supports both sync and async or just write the async version using aiohttp, and making aiohttp optional. So I sticked to the second option, knowing most users use sync version so that I had to be careful with my decisions.

**Reply by T-256** [reply]
Hello, 
`httpx` is both sync and async client.
For now I suggest use it for just async part by dropping `aiohttp` and its dependencies,
One reason that could prevent do this is using server part of `aiohttp` (do you? I didn't noticed).

`requests` replacement could be happening in future.

**Reply by T-256** [reply]
I searched codebase, few lines need changes.
I can send PR if you agree with this migration.

**Comment by Badiboy** [comment]
@coder2020official See neither need nor reason to make any fundamental changes.

**Comment by coder2020official** [comment]
True
